src/api/queues.js

```diff
@@ -33,12 +33,36 @@ function validateLocation(req, res, next) {
 router.get(
   '/',
   safeAsync(async (req, res, _next) => {
+    const {
+      locals: { userAuthz },
+    } = res
     const queuesResult = await Queue.scope(
       'defaultScope',
       'questionCount'
     ).findAll()
-    const queues = queuesResult.map(queue => queue.get({ plain: true }))
-    res.json(queues)
+    if (userAuthz.isAdmin) {
+      const queues = queuesResult.map(queue => queue.get({ plain: true }))
+      res.json(queues)
+    } else {
+      const courses = await Course.findAll()
+      const filteredCourseIds = courses
```

Comment:
Like the "get courses" endpoint, this is ideally logic that we could push into the query itself and let the query engine take care of all this.